x-pack/plugins/security_solution/public/common/components/alerts_treemap/lib/layers/index.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Datum } from '@elastic/charts';
+
+import { getFillColor } from '../chart_palette';
+import { getLabel } from '../labels';
+
+export interface DataName {
+  dataName: string;
+}
+
+export interface Path {
+  index: number;
+  value: string;
+}
+
+export interface FillColorDatum {
+  path?: Path[];
+}
+
+// common functions used by getLayersOneDimension and getLayersMultiDimensional:
+const valueFormatter = (d: number) => `${d}`;
+const groupByRollup = (d: Datum) => d.key;
+
+/**
+ * Extracts the first group name from the data representing the second group
+ */
+export const getGroupFromPath = (datum: FillColorDatum): string | undefined => {
+  const OFFSET_FROM_END = 2; // The offset from the end of the path array containing the group
+
+  const pathLength = datum.path?.length ?? 0;
+  const groupIndex = pathLength - OFFSET_FROM_END;
+
+  return Array.isArray(datum.path) && groupIndex > 0 ? datum.path[groupIndex].value : undefined;
```

Comment:
There will never be a situation where you have a one element array since this is extracted from the second group correct?

Reply:
> There will never be a situation where you have a one element array since this is extracted from the second group correct?

Summary:

That's correct, a one element array would fail the `groupIndex > 0`

Details:

The value of `groupIndex` is (mathematically) guaranteed to fail the  `groupIndex > 0` check if you have a one element array, because:

- `pathLength` would be `1` in this scenario
- the value of the `OFFSET_FROM_END` constant is `2`
- so `groupIndex` would evaluate to `-1`, when `const groupIndex = pathLength - OFFSET_FROM_END;` is executed

I added a new test for this case:

```ts
    it('returns undefined when path is an array with only one value', () => {
      expect(
        getGroupFromPath({
          path: [{ index: 0, value: '__null_small_multiples_key__' }],
        })
      ).toBeUndefined();
    });
```